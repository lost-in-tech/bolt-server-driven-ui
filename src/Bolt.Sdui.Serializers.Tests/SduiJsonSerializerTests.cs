using Bolt.Sdui.Serializers.Tests.Fixtures;

namespace Bolt.Sdui.Serializers.Tests;
public partial class SduiJsonSerializerTests : TestWithIocFixture
{
    private readonly ISduiJsonSerializer _sut;

    public SduiJsonSerializerTests(IocFixture fixture) : base(fixture)
    {
        _sut = GetRequiredService<ISduiJsonSerializer>();    
    }
}
