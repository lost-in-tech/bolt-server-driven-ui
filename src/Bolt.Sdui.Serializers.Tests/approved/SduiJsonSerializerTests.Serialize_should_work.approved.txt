{
  "statusCode": "ok",
  "layout": {
    "wide": {
      "type": "Stack",
      "direction": {
        "xs": "vertical",
        "sm": "vertical",
        "md": "vertical",
        "lg": "horizontal",
        "xl": "vertical"
      },
      "elements": [
        {
          "type": "Stack",
          "elements": [
            {
              "type": "Placeholder",
              "sectionName": "section-1"
            }
          ]
        },
        {
          "type": "Stack"
        },
        {
          "type": "Stack"
        }
      ]
    }
  },
  "sections": [
    {
      "name": "section-1",
      "element": {
        "type": "TestTextElement",
        "value": "hello world"
      }
    }
  ]
}